openmdao/core/system.py

```diff
@@ -3041,9 +3041,15 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
                 meta['size'] = int(meta['size'])  # make default int so will be json serializable
 
                 if src_name in abs2idx:
+                    indices = meta['indices']
                     meta = abs2meta_out[src_name]
-                    out[name]['distributed'] = meta['distributed']
-                    out[name]['global_size'] = meta['global_size']
+                    if indices is not None:
+                        # Index defined in this response.
+                        out[name]['global_size'] = len(indices) if meta['distributed'] \
+                            else meta['global_size']
```

Comment:
It looks like you are no longer setting `out['name']['distributed']` in this case, where we were before... 
Just want to check that this is not a potential problem...

Reply:
Good catch. I think we do want to keep 'distributed' set for both cases, so I will modify this.